pennylane/pulse/hardware_hamiltonian.py

```diff
@@ -266,19 +266,27 @@ def __init__(
         self,
         coeffs,
         observables,
+        reorder_fn=None,
         pulses: List["HardwarePulse"] = None,
         settings: Union["RydbergSettings", "TransmonSettings"] = None,
     ):
         self.settings = settings
         self.pulses = [] if pulses is None else pulses
+
+        def trivial_func(_, y):
+            return y
+
+        self.reorder_fn = trivial_func if reorder_fn is None else reorder_fn
         super().__init__(coeffs, observables)
 
     def __call__(self, params, t):
-        params = _reorder_parameters(params, self.coeffs_parametrized)
+        params = self.reorder_fn(params, self.coeffs_parametrized)
```

Comment:
I think you can also just run this line if `self.reorder_fn is not None`

Reply:
Also, with your current definition of `trivial_func`, `self.reorder_fn` would return `self.coeffs_parametrized` instead of `params`.